runtime/compiler/control/J9Options.cpp

```diff
@@ -1809,9 +1809,10 @@ J9::Options::fePreProcess(void * base)
 #if defined(TR_HOST_X86) || defined(TR_HOST_POWER) || defined(TR_HOST_S390)
    bool preferTLHPrefetch;
 #if defined(TR_HOST_POWER)
-   TR_Processor proc = TR_J9VMBase::getPPCProcessorType();
-   preferTLHPrefetch = proc >= TR_PPCp6 && proc <= TR_PPCp7;
+   // TODO: processor id() is not initialized at this point. Once we swtich to the new api this problem will go away
+   preferTLHPrefetch = (TR::Compiler->target.cpu.id() >= TR_PPCp6) && (TR::Compiler->target.cpu.id() <= TR_PPCp7);
 #elif defined(TR_HOST_S390)
+   // TODO: processor arch is not initialized at this point. Once we switch to the new api this problem will go away
```

Comment:
Good eye, more work is needed here. 

The problem here was that id() was not initialized, and it will break my test if I switched to the new api over here so I reverted back for this to use the old api. However what I've done here is incorrect.

I should set the id() with `TR_J9VMBase::getPPCProcessorType()` so that way I could use my new api here without breaking my test(the check between old api and new api)